api/api-doc/task_manager.json

```diff
@@ -0,0 +1,312 @@
+                "method":"POST",
+                "summary":"Abort running task and its descendants",
+                "type":"void",
+                "nickname":"abort_task",
+                "produces":[
+                   "application/json"
+                ],
+                "parameters":[
+                   {
+                      "name":"task_id",
+                      "description":"The id of a task to abort",
+                      "required":true,
+                      "allowMultiple":false,
+                      "type":"string",
+                      "paramType":"path"
+                   }
+                ]
+             }
+          ]
+       },
+       {
+        "path":"/task_manager/wait_task/{task_id}",
+        "operations":[
+           {
+              "method":"GET",
+              "summary":"Wait for a task to complete",
+              "type":"task_status",
+              "nickname":"wait_task",
+              "produces":[
+                 "application/json"
+              ],
+              "parameters":[
+                 {
+                    "name":"task_id",
+                    "description":"The id of a task to wait for",
+                    "required":true,
+                    "allowMultiple":false,
+                    "type":"string",
+                    "paramType":"path"
+                 }
+              ]
+           }
+        ]
+     },
+     {
+        "path":"/task_manager/wait_tasks",
+        "operations":[
+           {
+              "method":"POST",
+              "summary":"Wait for all tasks in the given list to complete",
+              "type":"long",
+              "nickname":"wait_tasks",
+              "produces":[
+                 "application/json"
+              ],
+              "parameters":[
+                 {
+                    "name":"tasks_ids",
+                    "description":"The ids of the tasks to wait for",
+                    "required":true,
+                    "allowMultiple":false,
+                    "type":"array",
+                    "items":{
+                       "type":"string"
+                    },
+                    "paramType":"query"
+                 }
+              ]
+           }
+        ]
+     },
+     {
+        "path":"/task_manager/get_task_ttl",
+        "operations":[
+           {
+              "method":"GET",
+              "summary":"How long tasks’ status are listed after they end, in seconds",
+              "type":"long",
+              "nickname":"get_task_ttl",
+              "produces":[
+                 "application/json"
+              ],
+              "parameters":[
+              ]
+           }
+        ]
+     },
+     {
+        "path":"/task_manager/set_task_ttl",
+        "operations":[
+           {
+              "method":"POST",
+              "summary":"Set the time in seconds to keep tasks status in memory after they end",
+              "type":"void",
+              "nickname":"set_task_ttl",
```

Comment:
This should be a live-update configuration item.

Reply:
I don't understand what you mean.

Reply:
This should be made a regular configuration item, added to `db::config` and marked with `liveness::LiveUpdate`  so that it can be updated run-time with .e.g. updates to `system.config` table, or the rest API corresponding endpoints.

Reply:
agreed